### DISCUSSION TITLE
How to remove field value using PATCH

### DISCUSSION THREAD

**Discussion by ameyapathare** [vitalik/django-ninja#675] [Q&A] [answered]
Hi! I am trying to write a PATCH API endpoint that can take a subset of fields and update a product object.  Say I have the following schema and endpoint:
```
class ProductUpdate(Schema):
    name: str = None
    description: str = None
    category_id: uuid.UUID = None

@api.patch("/{product_id}")
def update_product(request, product_id, payload: ProductUpdate):
    product = get_object_or_404(Product, id=product_id)
    print(payload.dict())
    for attr, value in payload.dict().items():
        if value:
            setattr(product, attr, value)
    product.save()
    return {"success": True}
```

If I just want to update the description, I can send this request body: `{"description": "new description"}`.  It looks like `payload.dict()` always includes `"name": None` and `"category_id": None`.

However, if I want to remove the category_id,  with a request body like `{"description": "new description", "category_id": None}`, I don't know how to know that the `"category_id": None` was passed in the request body versus was defaulted by django-ninja.  Is there any way to allow a PATCH request to remove values from models?

**Comment by vitalik** [comment]
@ameyapathare 

dict method [is from pydantic](https://docs.pydantic.dev/usage/exporting_models/#modeldict)

so you can try with `exclude_none`

```Python
for attr, value in payload.dict(exclude_none=True).items():
   ...
```

**Comment by sebastian-philipp** [comment, marked as answer]
https://fastapi.tiangolo.com/tutorial/body-updates/#partial-updates-with-patch might also be helpful

**Comment by ameyapathare** [comment]
Thanks both!  Knowing the dict method was from pydantic was the key unlock here.  The parameter I needed was "exclude_unset" (FastAPI documentation contained that as well).  Thanks!
